libs/datamodel/core/src/transform/ast_to_dml/lift.rs

```diff
@@ -212,204 +178,181 @@ impl<'a> LiftAstToDml<'a> {
-                            number_of_args,
-                            type_specification.unwrap().span,
-                        ));
-                    }
-                    if number_of_args > constructor._number_of_args + constructor._number_of_optional_args
-                        && constructor._number_of_optional_args > 0
-                    {
-                        return Err(DatamodelError::new_connector_error(
-                            &ConnectorError::from_kind(ErrorKind::OptionalArgumentCountMismatchError {
-                                native_type: x.parse().unwrap(),
-                                optional_count: constructor._number_of_optional_args,
-                                given_count: number_of_args,
-                            })
-                            .to_string(),
-                            type_specification.unwrap().span,
-                        ));
-                    }
-
-                    // check for compatibility with scalar type
-                    if !constructor.prisma_types.contains(&scalar_type) {
-                        return Err(DatamodelError::new_connector_error(
-                            &ConnectorError::from_kind(ErrorKind::IncompatibleNativeType {
-                                native_type: x.parse().unwrap(),
-                                field_type: scalar_type.to_string(),
-                                expected_types: constructor.prisma_types.iter().map(|s| s.to_string()).join(" or "),
-                            })
-                            .to_string(),
-                            type_specification.unwrap().span,
-                        ));
-                    }
-
-                    let parse_native_type_result = connector.parse_native_type(x, args);
-                    match parse_native_type_result {
-                        Err(connector_error) => Err(DatamodelError::new_connector_error(
-                            &connector_error.to_string(),
-                            type_specification.unwrap().span,
-                        )),
-                        Ok(parsed_native_type) => Ok((
-                            dml::FieldType::Scalar(scalar_type, None, Some(parsed_native_type)),
-                            vec![],
-                        )),
-                    }
-                } else {
-                    Ok((dml::FieldType::Scalar(scalar_type, type_alias, None), vec![]))
-                }
-            } else {
-                Ok((dml::FieldType::Scalar(scalar_type, type_alias, None), vec![]))
-            }
-        } else if self.db.ast().find_model(type_name).is_some() {
-            Ok((dml::FieldType::Relation(dml::RelationInfo::new(type_name)), vec![]))
-        } else if self.db.get_enum(type_name).is_some() {
-            Ok((dml::FieldType::Enum(type_name.clone()), vec![]))
-        } else {
-            self.resolve_custom_type(ast_field, checked_types)
+fn lift_scalar_field_type(
```

Comment:
This is mostly just moved in its own function, one 2-3 levels of indentation lower. No significant change to review.

Reply:
Five parameters. Would it still make sense to have this in the struct impl?